pkg/runtime/runtime.go

```diff
@@ -52,7 +55,35 @@ type kubernetes struct{}
 
 // NewRuntime - Initialize provider variable
 func NewRuntime(log *logging.Logger) {
-	Provider = &provider{log: log}
+	k8scli, err := clients.Kubernetes(log)
+	if err != nil {
+		log.Error(err.Error())
+		panic(err.Error())
+	}
+
+	// Identify which cluster we're using
+	restclient := k8scli.Client.CoreV1().RESTClient()
+	body, err := restclient.Get().AbsPath("/version/openshift").Do().Raw()
+
+	var cluster coe
+	switch {
+	case err == nil:
+		var kubeServerInfo kubeversiontypes.Info
+		err = json.Unmarshal(body, &kubeServerInfo)
+		if err != nil && len(body) > 0 {
+			log.Error(err.Error())
+			panic(err.Error())
+		}
+		log.Info("OpenShift version: %v", kubeServerInfo)
+		cluster = openshift{}
+	case kapierrors.IsNotFound(err) || kapierrors.IsUnauthorized(err) || kapierrors.IsForbidden(err):
```

Comment:
If we are accepting the scenario where we are not authorized to access the `/version/openshift` endpoint, should we not be ensuring we have permission to do work on a kubernetes cluster first?

I'm struggling to ask this question clearly. So I'm going to ask the same question again, worded differently, in the hopes I can communicate the idea. If we are not authorized / forbidden to access the `/version/openshift` **and** we are not authorized / forbidden to access `/version/kubernetes` (assuming there is a k8s version endpoint somewhere), should we fail?

Reply:
Agreed @djzager.  In ```/pkg/app/app.go```, I added a [failure case](https://github.com/openshift/ansible-service-broker/pull/574/files?diff=split#diff-637a60a871f8b69622bdfd3d3b2fcad1R194) on kubernetes because if this validation fails then we aren't connected to a cluster.

```go
        case kapierrors.IsNotFound(err) || kapierrors.IsUnauthorized(err) || kapierrors.IsForbidden(err):
                app.log.Error("the server could not find the requested resource")
                app.log.Error(err.Error())
                os.Exit(1)
```